### PR TITLE
New Keyboard: kaishi65

### PR DESCRIPTION
**Pull request by powerlemming**
## Description
The kaishi65 is a 65% keyboard from the newly created KBDClack company. The PCB is based on coseyfannitutti's MULLET design (thus the code is very similar) but adds support for an extra key and drops RGB support. (firmware created from new_keyboard.sh script)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/KBDClack/kaishi65/kaishi65.c [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    return process_record_user(keycode, record);
+}
+
+bool led_update_kb(led_t led_state) {
+    // put your keyboard LED indicator (ex: Caps Lock LED) toggling code here
+
+    return led_update_user(led_state);
+}
+*/
```
Comment (fauxpark) [inline keyboards/KBDClack/kaishi65/kaishi65.c:50]:
```suggestion
```

File: keyboards/KBDClack/kaishi65/keymaps/default/config.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2020 KBDClack
```
Comment (fauxpark) [inline keyboards/KBDClack/kaishi65/keymaps/default/config.h:1]: This file can be deleted.

File: keyboards/KBDClack/kaishi65/readme.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# kaishi65
+
+
+
```
Comment (fauxpark) [inline keyboards/KBDClack/kaishi65/readme.md:4]:
```suggestion

```

File: keyboards/KBDClack/kaishi65/readme.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+*(PCB based on coseyfannitutti's MULLET design)*
+
+* Keyboard Maintainer: [KBDClack](https://github.com/powerlemming)
+* Hardware Supported: The kaishi65 line of products (case + PCB)
+* Hardware Availability: www.kbdclack.com
+
+Make example for this keyboard (after setting up your build environment):
+
+    make KBDClack/kaishi65:default
```
Comment (fauxpark) [inline keyboards/KBDClack/kaishi65/readme.md:15]:
```suggestion
    make kbdclack/kaishi65:default
```

Please rename the folder to lowercase.

File: keyboards/KBDClack/kaishi65/config.h [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID 0xFEED
+#define PRODUCT_ID 0x0000
```
Comment (fauxpark) [inline keyboards/KBDClack/kaishi65/config.h:24]:
```suggestion
#define PRODUCT_ID 0x1A81
```

PID generated from first four digits of `echo -n "kaishi65" | shasum`

**Comment by powerlemming** [comment]
> 
> 
> Folder name should be lowercase.

I can't seem to get the folder name change pushed to GitHub. Any suggestion on how I should proceed ?

**Comment by fauxpark** [comment]
```
git mv KBDClack tmp
git mv tmp kbdclack
```

then commit & push.

File: keyboards/kbdclack/kaishi65/info.json [hunk 6]
```diff
@@ -0,0 +1,12 @@
+{
+    "keyboard_name": "kaishi65", 
+    "url": "https://github.com/powerlemming/KBDClack", 
+    "maintainer": "KBDClack", 
+    "width": 16, 
+    "height": 5, 
+    "layouts": {
+        "LAYOUT": {
```
Comment (zvecr) [inline keyboards/kbdclack/kaishi65/info.json:8]:
```suggestion
        "LAYOUT_65_ansi": {
```

File: keyboards/kbdclack/kaishi65/kaishi65.h [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+/* This is a shortcut to help you visually see your layout.
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+#define LAYOUT( \
```
Comment (zvecr) [inline keyboards/kbdclack/kaishi65/kaishi65.h:31]:
```suggestion
#define LAYOUT_65_ansi( \
```

File: keyboards/kbdclack/kaishi65/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+// Defines names for use in layer keycodes and the keymap
+enum layer_names {
+    _BASE,
+    _FN
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    /* Base */
+    [_BASE] = LAYOUT(
```
Comment (zvecr) [inline keyboards/kbdclack/kaishi65/keymaps/default/keymap.c:26]:
```suggestion
    [_BASE] = LAYOUT_65_ansi(
```

File: keyboards/kbdclack/kaishi65/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_GRV,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,  KC_PGUP,
+        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,   KC_PGDN,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                    KC_RALT, MO(_FN), KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT),
+
+    [_FN] = LAYOUT(
```
Comment (zvecr) [inline keyboards/kbdclack/kaishi65/keymaps/default/keymap.c:33]:
```suggestion
    [_FN] = LAYOUT_65_ansi(
```

File: keyboards/kbdclack/kaishi65/rules.mk [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```
Comment (zvecr) [inline keyboards/kbdclack/kaishi65/rules.mk:32]:
```suggestion

LAYOUTS = 65_ansi
```

**Comment by powerlemming** [comment]
Is there any changes still to be made ?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by noroadsleft** [comment]
For future reference, we recommend against committing to your `master` branch as you've done here, because pull requests from modified `master` branches can make it more difficult to keep your QMK fork updated. It is highly recommended for QMK development – regardless of what is being done or where – to keep your master updated, but **NEVER** commit to it. Instead, do all your changes in a branch (branches are basically free in Git) and issue PRs from your branches when you're developing.

There are instructions on how to keep your fork updated here:

[**Best Practices: Your Fork's Master: Update Often, Commit Never**](https://docs.qmk.fm/#/newbs_git_using_your_master_branch)

[Fixing Your Branch](https://docs.qmk.fm/#/newbs_git_resynchronize_a_branch) will walk you through fixing up your `master` branch moving forward. If you need any help with this just ask.

Thanks for contributing!

**Comment by powerlemming** [comment]
Quick question: why do I have an icon on my commit telling me that the travis-ci test failed ?

image

**Comment by noroadsleft** [comment]
> Quick question: why do I have an icon on my commit telling me that the travis-ci test failed ?

image

@mechmerlin's `65_ansi` keymap is bugged – it has an extra comma that shouldn't be there.

**Pull request merged**